armi/materials/material.py

```diff
@@ -464,11 +482,11 @@ def getLifeMetalConservativeFcciCoeff(self, Tk: float) -> float:
 
     def yieldStrength(self, Tk: float = None, Tc: float = None) -> float:
         r"""returns yield strength at given T in MPa"""
-        return self.p.yieldStrength
+        pass
 
     def thermalConductivity(self, Tk: float = None, Tc: float = None) -> float:
         r"""thermal conductivity in given T in K"""
-        return self.p.thermalConductivity
+        pass
```

Comment:
should be NotImplementedError? If not, explain in code comments.

Reply:
The `Material` class has `pass` methods, meant to be subclassed (and did before this PR).

For instance, `def applyInputParams()` is `pass` because if it was `NotImplementedError`, nearly all of the Materials would fail. The default functionality is to do nothing, because these `Material` methods get called willy-nilly by downstream code.

I tried, and `Material` is not an abstract class, but it's close. 

I can add some sort of comment to this affect.

Reply:
In retrospect, this is part of the "materials need to be more consistent" issue. Discussed more [here](https://github.com/terrapower/armi/discussions/767).

This PR aims to make Materials less stateful, and to not subclass deposit.

But I can see your issue.  This PR won't fix _everything_ about our `Material`s, and speaking to the broad user base, most issues people have come down to consistency.